cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -92,3 +106,17 @@ func parse(args []string) (*kong.Context, error) {
 
 	return parser.Parse(args)
 }
+
+func createLogger() *logrus.Logger {
+	logger := logrus.New()
+	logger.SetFormatter(&logrus.TextFormatter{})
+
+	file, err := os.OpenFile(logFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if err != nil {
+		logger.Info(errors.Wrapf(err, "Failed to log to file %s, using default of stderr", logFileName))
+	} else {
+		logger.Out = io.MultiWriter(os.Stderr, file)
+	}
```

Comment:
Two things here:

First, the code that I had copied from the Logrus website was wrong, if you look at the source of the New() function (do we still call them constructors?) you can see that it actually assigns os.Stderr (not stdout) by default so not only was it the wrong stream but we also didn't actually need to do any overwriting.

Second, and more importantly, we're now using a MultiWriter (TIL) to fork output to both stderr and the log file.